appium.js

```diff
@@ -73,12 +91,91 @@ function connectGetDefaultArgs () {
   });
 }
 
+function connectStartSession (win) {
+  ipcMain.on('start-session', () => {
+    let sessionWin = new BrowserWindow({width: 800, height: 600, webPreferences: {devTools: true}});
+    let sessionHTMLPath = path.resolve(__dirname, 'app', 'index.html#/session');
+    sessionWin.loadURL(`file://${sessionHTMLPath}`);
+    sessionWin.show();
+
+    // When you close the session window, kill the associated Appium session
+    let sessionWinID = sessionWin.webContents.id;
+    sessionWin.on('closed', () => {
+      killSession(sessionWinID);
+      sessionWin = null;
+    });
+
+    // When the main window is closed, terminate the appium session and close the session window
+    win.once('closed', () => {
+      // sessionWin.close();
+      sessionWin = null;
+    });
+
+    if (isDev) {
+      sessionWin.openDevTools();
+    }
+
+    sessionWin.webContents.on('context-menu', (e, props) => {
+      const {x, y} = props;
+
+      Menu.buildFromTemplate([{
+        label: 'Inspect element',
+        click () {
+          sessionWin.inspectElement(x, y);
+        }
+      }]).popup(sessionWin);
+    });
+  });
+}
+
+function connectCreateNewSession () {
+  ipcMain.on('appium-create-new-session', async (event, args) => {
+    const { desiredCapabilities, host, port, username, accessKey, https } = args;
+
+    let driver = sessionDrivers[event.sender.id] = wd.promiseChainRemote({
+      hostname: host,
+      port,
+      username,
+      accessKey,
+      https,
+    });
+
+    try {
+      let p = driver.init(desiredCapabilities);
+      event.sender.send('appium-new-session-successful');
+      await p;
+      event.sender.send('appium-new-session-ready');
+
+      // Listen for command requests
+      ipcMain.on('appium-client-command-request', async () => {
+        let type = 'source';
```

Comment:
presumably you're in the process of building this out to accommodate other request types?